src/Infrastructure/Service/Telemetry/TelemetryEventSenderFactory.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service\Telemetry;
+
+use InvalidArgumentException;
+
+class TelemetryEventSenderFactory
+{
+    /**
+     * @var bool
+     */
+    protected bool $isDebug;
+
+    /**
+     * @var string
+     */
+    protected string $dataLakeUrl;
+
+    /**
+     * @var iterable<\SprykerSdk\Sdk\Infrastructure\Service\Telemetry\TelemetryEventSenderInterface>
+     */
+    protected iterable $telemetryEventSenders;
+
+    /**
+     * @var bool
+     */
+    protected bool $isTelemetryEnabled;
+
+    /**
+     * @param bool $isDebug
+     * @param string $dataLakeUrl
+     * @param bool $isTelemetryEnabled
+     * @param iterable<\SprykerSdk\Sdk\Infrastructure\Service\Telemetry\TelemetryEventSenderInterface> $telemetryEventSenders
+     */
+    public function __construct(bool $isDebug, string $dataLakeUrl, bool $isTelemetryEnabled, iterable $telemetryEventSenders)
+    {
+        $this->isDebug = $isDebug;
+        $this->dataLakeUrl = trim($dataLakeUrl);
+        $this->isTelemetryEnabled = $isTelemetryEnabled;
+        $this->telemetryEventSenders = $telemetryEventSenders;
+    }
+
+    /**
+     * @return \SprykerSdk\Sdk\Infrastructure\Service\Telemetry\TelemetryEventSenderInterface
+     */
+    public function getTelemetryEventSender(): TelemetryEventSenderInterface
+    {
+        if (!$this->isTelemetryEnabled) {
+            return $this->createNullSender();
+        }
+
```

Comment:
please add logic  to make a decision based on the `transport` setting
something like:
```
//...
foreach ($this->transportRegistry->getTransports() as $transport) {
     if ($settingProvider->getTransport() === $transport->getName()) {
           return $transport;
     }
}
return $this->transportRegistry->getFallbackTransport();
```